NitroxPatcher/Patches/Dynamic/FootstepSounds_OnStep_Patch.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Emit;
+using FMOD.Studio;
+using HarmonyLib;
+using NitroxClient.GameLogic.FMOD;
+using NitroxModel.GameLogic.FMOD;
+using NitroxModel.Helper;
+using UnityEngine;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+public sealed partial class FootstepSounds_OnStep_Patch : NitroxPatch, IDynamicPatch
+{
+    private static readonly MethodInfo targetMethod = Reflect.Method((FootstepSounds t) => t.OnStep(default));
+
+    public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+    {
+        /*
+        From:
+
+            evt.setVolume(volume);
+
+        To:
+
+            evt.setVolume(CalculateVolume(volume, this, asset, xform));
+         */
+
+        return new CodeMatcher(instructions)
+               .MatchEndForward(
+                   new CodeMatch(OpCodes.Ldloc_1),
+                   new CodeMatch(OpCodes.Call, Reflect.Method((EventInstance evt) => evt.setVolume(default)))
+               )
+               .Insert(
+                   new CodeInstruction(OpCodes.Ldarg_0),
+                   new CodeInstruction(OpCodes.Ldloc_0),
+                   new CodeInstruction(OpCodes.Ldarg_1),
+                   new CodeInstruction(OpCodes.Call, Reflect.Method(() => CalculateVolume(default, default, default, default)))
+               )
+               .InstructionEnumeration();
+    }
+
+    // This method is called very often and should therefor be performant
+    private static float CalculateVolume(float originalVolume, FootstepSounds instance, FMODAsset asset, Transform xform)
+    {
+        // For exosuits only "event:/sub/exo/step" is used inside FootstepSounds
+        if (asset.path.Equals("event:/sub/exo/step", StringComparison.Ordinal) && (!Player.main.currentMountedVehicle || Player.main.currentMountedVehicle.gameObject != instance.gameObject))
```

Comment:
Path string could be moved on top of the file to also be used by ``stepSoundRadius``